aixplain/api/pipeline.py

```diff
@@ -0,0 +1,354 @@
+import uuid
+from dataclasses import dataclass, field, InitVar, asdict
+from typing import List, Tuple
+
+from aixplain.factories.pipeline_factory import PipelineFactory
+
+from .enums import (
+    NodeType,
+    DataType,
+    RouteType,
+    Operation,
+)
+from .base import Node, Link
+from .nodes import (
+    Asset,
+    Decision,
+    Script,
+    Input,
+    Output,
+    Route,
+    Router,
+    Reconstructor,
+    Segmentor,
+)
+
+
+@dataclass
+class Pipeline:
+    nodes: List[Node] = field(default_factory=list)
+    links: List[Link] = field(default_factory=list)
+    instance: InitVar[any] = None
+
+    def add_node(self, node: Node):
+        """
+        Add a node to the current pipeline.
+
+        This method will take care of setting the pipeline instance to the
+        node and setting the node number if it's not set.
+
+        :param node: the node
+        :return: the node
+        """
+        return node.attach(self)
+
+    def add_nodes(self, *nodes: Node) -> List[Node]:
+        """
+        Add multiple nodes to the current pipeline.
+
+        :param nodes: the nodes
+        :return: the nodes
+        """
+        return [self.add_node(node) for node in nodes]
+
+    def add_link(self, link: Link) -> Link:
+        """
+        Add a link to the current pipeline.
+        :param link: the link
+        :return: the link
+        """
+        return link.attach(self)
+
+    def input(self, data: str = None, *args, **kwargs) -> Node:
```

Comment:
Can we add these methods which create a node into a method called `PipelineNodeFactory`?

Reply:
I wouldn't do this as per our previous discussion on slack.